tests/rust-integration-tests/integration_test/src/tests/hostname/mod.rs

```diff
@@ -0,0 +1,53 @@
+use anyhow::{Context, Result};
+use oci_spec::runtime::{LinuxBuilder, ProcessBuilder, Spec, SpecBuilder};
+use test_framework::{Test, TestGroup, TestResult};
+
+use crate::utils::test_inside_container;
+
+fn create_spec(hostname: &str) -> Result<Spec> {
+    Ok(SpecBuilder::default()
+        .hostname(hostname)
+        .linux(
+            LinuxBuilder::default()
+                .readonly_paths(vec![
+                    "/proc/bus".to_string(),
+                    "/proc/fs".to_string(),
+                    "/proc/sys".to_string(),
```

Comment:
Why are we setting these specifically to ro? Is there anything in these tests that needs this?

Reply:
Without setting at least one read-only file, the test fails because of [this](https://github.com/containers/youki/blob/main/tests/rust-integration-tests/runtimetest/src/tests.rs#L14)

And if I leave the default configuration which is exactly the files I added and `/proc/sysrq-trigger` which when tried on my machine fails with a permission error [on](https://github.com/containers/youki/blob/main/tests/rust-integration-tests/runtimetest/src/tests.rs#L29-L32).

I am not sure if this is related to my setup (My machine is running a recent ubuntu version with a `5.15` kernel), and I am also not sure why there is a test to enforce the existence of read-only paths (I understand why do we need them, but it would've been nicer to be able to set the read_only_paths to `vec![]`).

Reply:
To be exact, this is the error I get when running with the default configuration.

```
container stderr was not empty, found : in readonly paths, error in testing read access for path /proc/sysrq-trigger : Os { code: 13, kind: PermissionDenied, message: "Permission denied" }
```

Reply:
Hmm, this seems to be issue with my implementation of ro paths tests then... I'll take a look at this later and update back. Thanks for the detailed info of errors :)

Reply:
Hey, so I checked on my system, and removing lines 9->18 works fine, I think there might be some issue with setup of sysreq-trigger on your side. Can you check the permissions of that file / if the file exists or not?

Reply:
Hey, I didn't add any vagrant-like base because primarily :
- I wanted these tests to run without needing extra stuff, and having minimal dependencies on env (not a good thing in hindsight)
- Up till now, there was no such issue reported by anyone who did update the tests. I personally have ran this on plain ubuntu and Pop OS and both worked fine. (again, not the best thing in hindsight)

That all said, if there is no issue with your setup, and the error is still present, we would implement a vagrant based solution, as already done for containerd tests :+1:

Reply:
@YJDoc2 update: the file does exist with permissions `-w------- | 200`, so it is not readable, the test will always fail on my machine regardless.

Reply:
Interestingly, it now also fails on my machine, so I suspect I was using old build of tests when I reported that it works. However, you are correct in this, and can I ask you to do following changes?
- In readonly paths test, silently return instead of showing error when no paths are set (in both cases)
- in hostname test, set ro paths to either None or empty vector

I feel this is a better solution than setting some specific ro paths in a hostname test and explaining via comment, as well as will also help in future tests
WDYT?